index.html

```diff
@@ -215,8 +216,215 @@ <h3>WebSocket Re-use</h3>
+          successfully reading the value of the corresponding property it MUST send a <a
+            href="propertyReading"><code>propertyReading</code></a> message in response, containing its current value.
+        </p>
+      </section>
+      <section id="propertyReading">
+        <h4><code>propertyReading</code></h4>
+        <p>To notify a <a>Consumer</a> of the value of a <a>Property</a>, a <a>Thing</a> MUST send a
+          <a href="#websocket-messages">message</a> to the <a>Consumer</a> which contains the following members:
+        </p>
+        <table class="def">
+          <caption>Members of a <code>propertyReading</code> message</caption>
+          <thead>
+            <tr>
+              <th>Member</th>
+              <th>Type</th>
+              <th>Assignment</th>
+              <th>Description</th>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td><code>messageType</code></td>
+              <td>string</td>
+              <td>"propertyReading"</td>
+              <td>A string which denotes that this message is notifying a <a>Consumer</a> of the value of a
+                <a>Property</a>.
+              </td>
+            </tr>
+            <tr>
+              <td><code>name</code></td>
+              <td>string</td>
+              <td>Mandatory</td>
+              <td>The name of the <a>Property</a> being read.</td>
+            </tr>
+            <tr>
+              <td><code>value</code></td>
+              <td>any</td>
+              <td>Mandatory</td>
+              <td>The current value of the <a>Property</a> being read, with a format conforming to the data schema of
+                the
+                corresponding PropertyAffordance in the <a>Thing Description</a>.</td>
+            </tr>
+            <tr>
+              <td><code>timestamp</code></td>
+              <td>string</td>
+              <td>Optional</td>
+              <td>A timestamp in date-time format [[RFC3339]] set to the time the property reading took place.</td>
+            </tr>
+          </tbody>
+        </table>
+        <p>If a <code>propertyReading</code> message is sent in response to a <a
+            href="#readProperty"><code>readProperty</code></a>, <code>readAllProperties</code>,
+          <code>observeProperty</code> or <code>observeAllProperties</code> message which contained a
+          <code>correlatonID</code> member, then the response message SHOULD also include a <code>correlationID</code>
+          member with the same value.
+        </p>
+        <pre class="example" title="Property reading message (Thing to Consumer)">
+          {
+            "thingID": "https://mythingserver.com/things/mylamp1",
+            "messageID": "79057736-3e0e-4dc3-b139-a33051901ee2",
+            "messageType": "propertyReading",
+            "name": "on",
+            "value": true,
+            "timestamp": "2024-01-13T23:20:50.52Z",
```

Comment:
I think we had a dedicated discussion on this in the Strawman proposal. However, I believe we need to two timestamp in the final version:

* Creation Timestamp: Indicates when the message was created.
* Update Timestamp: (Optional) Reflects when the property was last updated/changed. This can be omitted if the device cannot provide it.

Currently, I understand that the timestamp represents when the property was read (the message was created), but it does not indicate when the property was updated/changed. 
If we add both timestamp, should we then use more clear names for the timestamps?

Reply:
I see what you mean, kind of like `Date` vs. `Last-Modified` in HTTP response headers.

I'm not sure what we'd call them. Maybe `timestamp` and `lastModifiedTimestamp`...

I'm curious whether any existing protocol bindings include a last modified timestamp in property readings because this is obviously not something that all Things can provide and almost (but not quite) feels like a separate property (e.g. asking a light switch when it was last turned on). @egekorkan @relu91?

Reply:
I definitely think that a lastUpdated or lastChanged timestamps are not separate properties, because you might want to have these timestamps on all properties. Either it's additional metadata of a property or the property would need to be a JSON object.
Otherwise it would lead to an explosion of properties.

Reply:
Wrt lastModified vs lastUpdated, if a property has been 'refreshed' with the same value, I consider this an update but not a modification. (for example in case a device refreshes property values at least once a day to confirm the value is still up to date).

Reply:
Exactly, there is a semantic difference between lastUpdated and lastModified/lastChanged.